packages/aws-cdk-lib/cloud-assembly-schema/lib/cloud-assembly/context-queries.ts

```diff
@@ -92,6 +92,13 @@ export interface AmiContextQuery {
    * Filters to DescribeImages call
    */
   readonly filters: {[key: string]: string[]};
+
+  /**
+   * Adds an additional discriminator to the `cdk.context.json` cache key.
+   *
+   * @default - no additional cache key
+   */
+  readonly additionalCacheKey?: string;
```

Comment:
When the context provider sends out the query, it also picks the context cache key. We can just mix the additional string into the context key; it never needs to make it to the cloud assembly as a separate field.

Reply:
I believe the reason I did this was so that when the queries are constructed like so:

```ts
const attributes: cxapi.KeyContextResponse = ContextProvider.getValue(scope, {
      provider: cxschema.ContextProvider.KEY_PROVIDER,
      props: {
        aliasName: options.aliasName,
        additionalCacheKey: options.additionalCacheKey,
      } as cxschema.KeyContextQuery,
      ...
});
```
it would be surfaced as a valid part of props. You're saying this is not necessary?